test/flush_worker_test.cc

```diff
@@ -0,0 +1,130 @@
+#include <functional>
+#include <thread>
+
+#include "external/envoy/source/common/common/random_generator.h"
+#include "external/envoy/source/common/event/dispatcher_impl.h"
+#include "external/envoy/source/common/runtime/runtime_impl.h"
+#include "external/envoy/source/common/stats/isolated_store_impl.h"
+#include "external/envoy/test/mocks/api/mocks.h"
+#include "external/envoy/test/mocks/local_info/mocks.h"
+#include "external/envoy/test/mocks/protobuf/mocks.h"
+#include "external/envoy/test/mocks/stats/mocks.h"
+#include "external/envoy/test/mocks/thread_local/mocks.h"
+#include "external/envoy/test/test_common/simulated_time_system.h"
+
+#include "client/flush_worker_impl.h"
+
+#include "gtest/gtest.h"
+
+namespace Nighthawk {
+namespace Client {
+
+using namespace testing;
+
+class FlushWorkerTest : public Test {
+public:
+  FlushWorkerTest() : thread_factory_(Envoy::Thread::threadFactoryForTest()) {
+    dispatcher_ = new NiceMock<Envoy::Event::MockDispatcher>();
```

Comment:
`std::make_unique`?

Reply:
mock_api_ will take the ownership of dispatcher_ by 
`EXPECT_CALL(mock_api_, allocateDispatcher_(_, _)).WillOnce(Return(dispatcher_));`
https://github.com/envoyproxy/envoy/blob/e8aab1b4f14baf7d7cf89297e0422dac6c7aac8e/test/mocks/api/mocks.cc#L23

Reply:
I see, makes sense then, sorry about the noise.